### PR TITLE
Serialize l20n to object

### PR DESCRIPTION
**Pull request by stasm**
https://bugzilla.mozilla.org/show_bug.cgi?id=1182406

### CODE REVIEW

**Comment by stasm** [comment]
@zbraniecki r?

File: src/bindings/gaiabuild/serialize.js [hunk 1]
```diff
@@ -30,6 +43,14 @@ export function serializeEntries(lang, langEntries, sourceEntries) {
   return [errors, entries];
 }
 
+function extend(into, from) {
+  for (let key in from) {
+    // overwrite existing keys for reduceRight
+    into[key] = from[key];
```
Comment (zbraniecki) [inline src/bindings/gaiabuild/serialize.js]: from is a keyword in ES6. I doubt it'll be a syntax error, but maybe use a different variable name?

File: src/bindings/gaiabuild/serialize.js [hunk 2]
```diff
@@ -2,7 +2,20 @@
...
+export function serializeContext(ctx, lang) {
+  const cache = ctx._env._resCache;
+  return ctx._resIds.reduceRight(([errorsSeq, entriesSeq], cur) => {
+    const sourceRes = cache[cur + 'en-USapp'];
+    const langRes = cache[cur + lang.code + lang.src];
+    const [errors, entries] = serializeEntries(
+      lang,
+      langRes instanceof L10nError ? {} : langRes,
+      sourceRes instanceof L10nError ? {} : sourceRes);
+    return [errorsSeq.concat(errors), extend(entriesSeq, entries)];
```
Comment (zbraniecki) [inline src/bindings/gaiabuild/serialize.js]: Is there a way to use Object.assign here instead of custom extend?

Comment (stasm) [inline src/bindings/gaiabuild/serialize.js]: Ah, perfect, just what I needed! Thanks.

**Pull request merged**